skops/card/_model_card.py

```diff
@@ -7,55 +7,65 @@
 import skops
 
 
-def _extract_estimator_config(model):
-    """Extracts estimator configuration and renders them into a vertical table.
-
-    Parameters
-    ----------
-        model (estimator): scikit-learn pipeline or model.
-
-    Returns
-    -------
-    str:
-        Markdown table of hyperparameters.
-    """
-    hyperparameter_dict = model.get_params(deep=True)
-    table = "| Hyperparameters | Value |\n| :-- | :-- |\n"
-    for hyperparameter, value in hyperparameter_dict.items():
-        table += f"| {hyperparameter} | {value} |\n"
-    return table
-
-
-def create_model_card(
-    model,
-    card_data,
-    **card_kwargs,
-):
-    """Creates a model card for the model and saves it to the target directory.
-
-    Parameters:
-    ----------
-    model: estimator
-        scikit-learn compatible estimator.
-    card_data: CardData
-        CardData object.
-    card_kwargs:
-        Card kwargs are information you can pass to fill in the sections of the
-        card template, e.g. model_description, citation_bibtex, get_started_code.
-    """
-    ROOT = skops.__path__
-    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
-    hyperparameter_table = _extract_estimator_config(model)
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
```

Comment:
Are we going to allow them to use an alternative `jinja` template?

If we do, it can be a separate PR though.

Reply:
Sounds like a good idea, agree that it can be another PR.

Reply:
From what I see the alternative template can be massed via `card.add(template_path="...")` which I find not super intuitive. I think in the constructor would make more sense. I left a comment at https://github.com/skops-dev/skops/pull/37/files#diff-1ba607fa6851b439c95dfe0e29c812a84e15748682b5c37b049b1fa8c474ba75R113